general/solomiia_tymoshchuk/lecture2/_1/Calculator.java

```diff
@@ -0,0 +1,73 @@
+package general.solomiia_tymoshchuk.lecture2._1;
+
+import java.util.Scanner;
+
+/**
+ * Define a class that represents a calculator.
+ * The class should provide methods that accept another value and
+ * perform addition, subtraction, multiplication and division
+ * operations on a value stored in a calculator instance.
+ * Divides by a zero should be forbidden and ignored.
+ * The class should also provide a method to get a current value.
+ * The class should work with both integer and double numbers (ignore
+ * roundoff errors). Assume all operation results fit into the range of
+ * values for a current value type.
+ **/
+
+public class Calculator {
+    private double number;
+
+    public Calculator(double number){
+        this.number = number;
+    }
+
+    void validateNumber(double number1, String operator){
+        if(number1 == 0 && operator.equals("/")){
+            throw new ArithmeticException("Divided by zero operation cannot possible");
+        }
+    }
+    void calculateNumbersSwitchCase(double number1, String operator) {
+        try{
+        validateNumber(number1, operator);
+        }
+        catch (ArithmeticException e){
+            System.out.println("Divided by zero operation cannot possible");
+            return;
+        }
+            switch (operator) {
+                case "+":
+                    System.out.println(number + number1);
```

Comment:
the value stored in the class should be changed after the operation; currently, it remains the same

Reply:
> the value stored in the class should be changed after the operation; currently, it remains the same

as far as i understand, here is separate methods for each operation required?

Reply:
Not exactly: each operation needs to modify the value of an instance variable `number`. A simple example: let `number` = 5, and you call the method `calculateNumbersSwitchCase` with `number1` = 3 and and `operator` = +.  The method will output the correct value `8`, but the value of the `number` will remain `5`. You need to do both - output a value and change a value, stored in the variable `number`. So, after example operation a value of the variable should be also 8. The same should be applied for other operations as well.